onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -8380,8 +8380,21 @@ def aten__unique(
 ) -> tuple[TensorType, TensorType]:
     """_unique(Tensor self, bool sorted=True, bool return_inverse=False) -> (Tensor, Tensor)"""
 
-    unique_values, _, inverse_indices, _ = op.Unique(self, axis=None, sorted=True)
+    unique_values, indices, inverse_indices, _ = op.Unique(self, axis=None, sorted=True)
+    # HACK: force indices to be in the graph so that it gets a name during optimization
```

Comment:
I suggest removing all hacks. I will go fix what's necessary where the bug is. We are also moving to prefer `trace_only=True` for new functions so if you can include the flag in `@torch_op` that would be awesome.

Reply:
That would be awesome. The hacks are definitely getting out of hand. I'll wait for that fix so that I can continue to test with this locally.

Reply:
Do you have a short script handy that will reproduce the error?

Reply:
```python
if __name__ == '__main__':
    import logging
    import torch
    import numpy as np
    import onnx
    import onnxruntime as ort
    for i in range(16):
        sorted = bool(i & 1)
        return_inverse = bool((i & 2) > 1)
        return_counts = bool((i & 4) > 1)
        dim = 0 if bool((i & 8) > 1) else None

        def test_function(
                x: torch.Tensor,
                s: bool = sorted,
                ri: bool = return_inverse,
                rc: bool = return_counts,
                d: int | None = dim) -> Any:
            result = torch.unique(
                x,
                sorted=s,
                return_inverse=ri,
                return_counts=rc,
                dim=d)
            if ri or rc:
                result = result[1]
            return result

        onnx_program = torch.onnx.dynamo_export(
            test_function,
            torch.arange(10),
            export_options=torch.onnx.ExportOptions(
                dynamic_shapes=True,
                diagnostic_options=torch.onnx.DiagnosticOptions(
                    verbosity_level=logging.DEBUG)))
        onnx_program.save("torch_unique.onnx")
        onnx_inputs = onnx_program.adapt_torch_inputs_to_onnx(torch.arange(10))
        onnx_outputs = onnx_program(*onnx_inputs)
        loaded_onnx_program = onnx.load("torch_unique.onnx")
        onnx.checker.check_model(loaded_onnx_program)
        ort_session = ort.InferenceSession("torch_unique.onnx")
        inputs = np.random.randint(0, 10, 10)
        print(f"Inputs: {inputs}")
        outputs = ort_session.run(None,
                                  {"l_x_": inputs})
        print(f"Outputs: {outputs}")
    print("Success")
```

Reply:
Oh, you should also test using the nightly release of PyTorch with the changes in https://github.com/pytorch/pytorch/pull/126561.

Reply:
Is `trace_only=True` expected to require significant changes to the way one implements an op? It appears that enabling the flag breaks passing a value to `op.ConstantOfShape` and also breaks indexing a shape.

For example,  `op.ConstantOfShape([0], value=[0])` must become `op.Cast(op.ConstantOfShape([0]), to=INT64.dtype)`, and ` output_size[dim]` must become `op.Slice(output_size, [dim], [dim+1])`.

Reply:
Your observation is correct. This may be the case because the gaps in implementation we have. Bridging the gaps is in our roadmap but is not the highest priority for the team.